crds/virtualmachineoperations.yaml

```diff
@@ -64,8 +64,41 @@ spec:
                       enum: ["Restart", "Stop"]
                   required: ["virtualMachineName"]
             status:
+              description: |
+                The latest observed state of the VirtualMachineOperation resource.
               type: object
               properties:
+                conditions:
+                  description: |
+                    The latest detailed observations of the VirtualMachineOperation resource.
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      lastProbeTime:
+                        description: Last time the condition was checked.
+                        format: date-time
+                        type: string
+                      lastTransitionTime:
+                        description: Last time the condition transit from one status to another.
+                        format: date-time
+                        type: string
+                      message:
+                        description: Human readable message indicating details about last transition.
+                        type: string
+                      reason:
+                        description: (brief) reason for the condition's last transition.
+                        type: string
+                      status:
+                        description: Status of the condition, one of True, False, Unknown.
+                        type: string
+                        enum: ["True", "False", "Unknown"]
+                      type:
+                        description: Type of condition.
+                        type: string
+                    required:
+                      - status
+                      - type
```

Comment:
The reason must be a required field

Reply:
Only VMClass has additional fields in required:
    "lastTransitionTime",
    "message",
    "reason",

Reply:
Add required fields, use Builder and conditions.SetCondition